packages/connect/src/Detector.ts

```diff
@@ -41,8 +42,20 @@ export class Detector {
       throw new Error(`No known Chain was detected and no chainSpec was provided. Either give a known chain name ('${Object.keys(this.#chainSpecs).join('\', \'')}') or provide valid chainSpecs.`)
     }
     await provider.connect();
+    let derives, initWasm, metadata, registry, rpc, signedExtensions, signer, source;
```

Comment:
Why explicitly name all these properties and copy them? Wouldn't it be neater to just use Object.assign?

Reply:
You mean something like this:
```
return await ApiPromise.create({ provider, ...Object.assign({}, options) });
```
?

Reply:
you want `provider` not to be overridden even if its supplied by the user so:

```javascript
return await ApiPromise.create(Object.assign(options, {provider}));
```

Reply:
That is not correct. If options are not provided this will lead to error